scheduler/src/cook/mesos/constraints.clj

```diff
@@ -160,10 +157,12 @@
       (let [vm-resources (.getCurrAvailableResources target-vm)
             vm-attributes (get-vm-lease-attr-map vm-resources)
             [passes? reason] (job-constraint-evaluate constraint vm-resources vm-attributes
-                                                      (into (vec (.getRunningTasks target-vm))
-                                                              (mapv (fn [^com.netflix.fenzo.TaskAssignmentResult result]
-                                                                      (.getRequest result))
-                                                                     (.getTasksCurrentlyAssigned target-vm))))]
+                                                      ;; This function is called for each job considered, for each host available.
+                                                      ;; It was optimized to reduce memory allocations because it accounted for ~10% of all allocs
+                                                      (persistent! (reduce (fn [tasks ^com.netflix.fenzo.TaskAssignmentResult result]
```

Comment:
I think it is fine, concat is only dangerous if there is a chain of concats, i.e. `(concat (concat (concat a b) c) d)`